src/views/MoviePage.vue

```diff
@@ -1,13 +1,160 @@
 <script>
 import { defineComponent } from 'vue';
 
+import { getMovieDetails } from '../api/monteCinemaApi';
+
+import BaseTag from '@/components/global/BaseTag.vue';
+
 export default defineComponent({
-  name: 'MoviePage'
+  name: 'MoviePage',
+  components: {
+    BaseTag
+  },
+  data() {
+    return {
+      movie: {
+        id: null,
+        title: '',
+        genre: '',
+        length: null,
+        release_date: '',
+        description: '',
+        poster_url: ''
+      }
+    };
+  },
+  computed: {
+    movieId() {
+      return this.$route.params.movieId;
+    },
+    movieDuration() {
+      const hours = Math.floor(this.movie.length / 60);
+      const minutes = `0${this.movie.length % 60}`.slice(-2);
+      return `${hours}h ${minutes}min`;
+    },
+    movieGenre() {
+      return this.movie.genre.name;
+    },
+    movieReleaseYear() {
+      const releaseDate = new Date(this.movie.release_date);
+      return releaseDate.getFullYear();
+    }
+  },
+  watch: {
+    movieId: {
+      immediate: true,
+      async handler(movieId) {
+        try {
+          this.movie = await getMovieDetails(movieId);
+        } catch (error) {
+          console.error(error);
```

Comment:
hmm.. do you expect users to look at the console? 😁  if you don't have better idea for now and just want to move forward - even `alert` is better than nothing 😉

Reply:
sure thing! It's not ideal but definitely better than nothing :)